### PR TITLE
Implemented upcoming-garbage-recycling-dates contract

### PR DESCRIPTION
**Pull request by mrosack**
Hey Mike - here's my implementation of the common garbage/recycling contract.  I'm a relative noob to python, so let me know if there's anything I did that could be done better.  Thanks!

### CODE REVIEW

File: appletonapi.py [hunk 1]
```diff
@@ -51,6 +51,16 @@ def sanitizeinputwords(rawinput):
 
 class PropertyHandler(webapp2.RequestHandler):
     def get(self, propkey):
+        result = self.execute(propkey)
```
Comment (mikeputnam) [inline appletonapi.py]: Thank you for the cleanup/consolidation here.

File: appletonapi.py [hunk 2]
```diff
@@ -106,16 +114,23 @@ def get(self, propkey):
...
+                return { 'propdetail' : datagroups }
             except urllib2.HTTPError, response:
-                self.response.out.write('error - Scrape: ' + str(response))
+                return { 'error' : 'error - Scrape: ' + str(response) }
 
 
 class SearchHandler(webapp2.RequestHandler):
     def get(self):
-        search_input = self.request.get('q')
+        result = self.execute(self.request.get('q'), str(self.request.headers['User-Agent']))
```
Comment (mikeputnam) [inline appletonapi.py]: And here.

File: appletonapi.py [hunk 3]
```diff
@@ -182,12 +197,63 @@ def get(self):
...
+                    result.append( { 'collectionType' : 'trash', 'collectionDate' : today_string } )
+                
+                if cur_date.strftime('%m-%d-%Y') == split_recycling_day[1].strip():
+                    result.append( { 'collectionType' : 'recycling', 'collectionDate' : today_string } )
+                    found_recycling = True
+                    
+                cur_date += timedelta(days=1)
+                failsafe_count += 1
+
+        self.response.headers["Content-Type"] = "application/json"
```
Comment (mikeputnam) [inline appletonapi.py]: But this one is inline?

Comment (mrosack) [inline appletonapi.py]: Good point - the main reason I broke those out was to make it easier for me to call them, but you're right that once I've started a new pattern I should keep it going... I'll make that fix (should I refactor crimes as well)?

File: appletonapi.py [hunk 4]
```diff
@@ -251,7 +317,8 @@ def get(self):
         ('/', MainHandler),
         (r'/property/(\d+)', PropertyHandler),
         ('/search', SearchHandler),
-        ('/crimes', CrimesHandler)
+        ('/crimes', CrimesHandler),
+        ('/garbageCollection', GarbageCollectionHandler)
```
Comment (mikeputnam) [inline appletonapi.py]: I'm not really fan of anything but lowercase endpoints.  Do you feel strongly about this?  *nix/*BSD servers treat urls case-sensitively whereas Windows servers do not care and are case-INsensitive.

Comment (mrosack) [inline appletonapi.py]: No problem - i'll change it to lowercase.

File: appletonapi.py (hunk 3 above)
Comment (mikeputnam) [inline appletonapi.py]: If you'd like to do the crimes one too that'd be excellent!

**Comment by mikeputnam** [comment]
Woo! Thanks so much @mrosack !

Closes issue #19

**Pull request merged**